src/ripple/overlay/impl/Manifest.cpp

```diff
@@ -134,35 +134,21 @@ ManifestCache::configValidatorKey(
     auto const words = beast::rfc2616::split(line.begin(), line.end(), ' ');
 
     if (words.size () != 2)
-    {
         Throw<std::runtime_error> ("[validator_keys] format is `<key> <comment>");
-    }
 
-    Blob key;
-    if (! Base58::decodeWithCheck (words[0], key))
-    {
+    auto const masterKey = parseBase58<PublicKey>(
+        TokenType::TOKEN_ACCOUNT_PUBLIC, words[0]);
+
+    if (!masterKey)
         Throw<std::runtime_error> ("Error decoding validator key");
-    }
-    if (key.size() != 34)
-    {
-        Throw<std::runtime_error> ("Expected 34-byte validator key");
-    }
-    if (key[0] != TOKEN_NODE_PUBLIC)
-    {
-        Throw<std::runtime_error> ("Expected TOKEN_NODE_PUBLIC (28)");
-    }
-    if (key[1] != 0xED)
-    {
-        Throw<std::runtime_error> ("Expected Ed25519 key (0xED)");
-    }
 
-    auto const masterKey = PublicKey (Slice(key.data() + 1, key.size() - 1));
-    std::string comment = std::move(words[1]);
+    if (publicKeyType(*masterKey) != KeyType::ed25519)
+        Throw<std::runtime_error> ("Validator key must use Ed25519");
 
-    if (journal.debug) journal.debug
-        << toBase58(TokenType::TOKEN_NODE_PUBLIC, masterKey) << " " << comment;
+    JLOG (journal.debug) << "Loaded key: " <<
+        toBase58(TokenType::TOKEN_NODE_PUBLIC, *masterKey);
```

Comment:
I don't know how expensive `toBase58()` is, but you have `masterKey` in text form already as `word[0]`.  You could consider writing `word[0]` to the log instead of calling `toBase58()`.  Your call.